[web] Alliance Page  (PART A - client side)

aaronmgdr opened the pull request:
### Description

**Creation of New Top Level Page For the Celo Alliance.** 

- add darken on hover to transparent header
- alliance members by category and ability to filter them
- apply form send data to airtable (and mailing list)
- change version to 4.0, to keep inline with figma versions
- create checkbox which use a real html input type checkbox, while appearing nice. 


### Related issues
DEPENDS ON  https://github.com/celo-org/celo-monorepo/pull/3048

### Tested

<img width="1425" alt="Screen Shot 2020-03-08 at 2 07 54 PM" src="https://user-images.githubusercontent.com/3814795/76171272-7ebad580-6146-11ea-93aa-9a4fe3afebfd.png">
<img width="1426" alt="Screen Shot 2020-03-08 at 2 08 21 PM" src="https://user-images.githubusercontent.com/3814795/76171274-81b5c600-6146-11ea-86e7-05ac1aa0b5f8.png">
<img width="1358" alt="Screen Shot 2020-03-08 at 2 08 37 PM" src="https://user-images.githubusercontent.com/3814795/76171276-824e5c80-6146-11ea-9a4d-f02e98d0ad0d.png">
<img width="1301" alt="Screen Shot 2020-03-08 at 2 08 47 PM" src="https://user-images.githubusercontent.com/3814795/76171277-837f8980-6146-11ea-8a46-6d74aed52e81.png">

<img width="1287" alt="Screen Shot 2020-03-08 at 2 08 54 PM" src="https://user-images.githubusercontent.com/3814795/76171278-84182000-6146-11ea-83b5-41cb7dfc476c.png">
<img width="1380" alt="Screen Shot 2020-03-08 at 2 09 02 PM" src="https://user-images.githubusercontent.com/3814795/76171279-84b0b680-6146-11ea-84ed-78afa63a0172.png">
<img width="1197" alt="Screen Shot 2020-03-08 at 2 09 10 PM" src="https://user-images.githubusercontent.com/3814795/76171281-85494d00-6146-11ea-875e-7044bd466785.png">
<img width="1329" alt="Screen Shot 2020-03-08 at 2 09 27 PM" src="https://user-images.githubusercontent.com/3814795/76171282-85494d00-6146-11ea-8394-75e2797fb859.png">

### Other changes

* Switched Header to use screenSize Context instead of the `<Responsive>` as Responsive did not play well with `<Hoverable>` 

* make use of Next.js `api` (instead of making index.ts, place route handle files in `pages/api` 

* move validationErrors trans to common for easier sharing

* start enforcing style that local links are defined in `pagePaths` while links to other sites are stored in `CeloLinks` 




- Fixes https://github.com/celo-org/celo-labs/issues/191

aaronmgdr committed:
💫 magic

aaronmgdr committed:
Woo woo

aaronmgdr committed:
Collecting collectivists

aaronmgdr committed:
Celo collective benefits area

aaronmgdr committed:
💖 testing out text sparkle animation

Commit:
rename collective -> alliance

Commit:
bigger badder sweeper

Commit:
Merge branch 'master' into aaronmgdr/collective

# Conflicts:
#	packages/web/src/header/Header.3.tsx

Commit:
refactor menu links

Commit:
fixes menu for tablet devices

Commit:
collective --> alliance

Commit:
Header / Menu fix and refactor

replace custom watching of resize with use of withScreenSize context
add Hoverable to menu
Fix issue with scrolling down past open mobile menu

Commit:
fix hover to be full width
replace responsive with screen size context

Commit:
wIP

Commit:
Merge branch 'master' into aaronmgdr/collective

# Conflicts:
#	packages/web/src/shared/Footer.tsx

Commit:
that rainbow

Commit:
add start of signup form

Commit:
Pretty alliance form

Commit:
improve spacing and mobile view

Commit:
Affirmations area of Alliance page

Commit:
alliance affirmations

Commit:
get on my collective

Commit:
dont require image import

Commit:
MAKE IT optional

Commit:
make dark mode declaration optional

Commit:
Alliance bottom mirror of top cover

Commit:
fetch alliance members from airtable

use next's api routes feature

movie error response to its own file

Commit:
update server-env-config

add alliance table key

Commit:
loads alliance members and renders them to page someone like the design

Commit:
add members link and open graph data

Commit:
add dropdown for filters

Commit:
working filtering of allies by function

Commit:
rename variable to be more clear

Commit:
add high quality icons

Commit:
add test

Commit:
WIP WIP

Commit:
Merge branch 'master' into aaronmgdr/collective

# Conflicts:
#	packages/web/src/_page-tests/__snapshots__/about.test.tsx.snap
#	packages/web/src/_page-tests/__snapshots__/community.test.tsx.snap
#	packages/web/src/_page-tests/developers/__snapshots__/Index.test.tsx.snap
#	packages/web/src/_page-tests/experience/__snapshots__/typography.test.tsx.snap
#	packages/web/src/_page-tests/validators/__snapshots__/index.test.tsx.snap
#	packages/web/src/community/connect/ArticlesSection.tsx
#	packages/web/src/shared/ConnectionFooter.tsx
#	packages/web/src/shared/menu-items.ts

Commit:
add url

Commit:
include url

Commit:
very WIP

Commit:
addd big quotes

Commit:
adds checkbox !

Commit:
move contributors to api folder

Commit:
rename Invest to Grow

Commit:
Form now sends data to server
string is to restrictive as sometimes form could have a boolean, string | boolean causes the need to do as string everywhere which is unpleasent

Commit:
Send Data from form to Airtable and the CRM.

Fix status code, use created instead of no content

Commit:
tighten up menu space so alliance fits

Commit:
add urls for reports

Commit:
set logo width drynamically

Commit:
woo variable width images

Commit:
SOME clean up (abort controller WIP)

Commit:
dont set State if component has unmounted

Commit:
Add Insure Type

Commit:
Make menu dark not black
Member logos should be span width space around
Double vertical space on each logouts
center pinwheell mobile + still mode for lower area
alliance before community on header

Commit:
full width on mobile

Commit:
fix full width button

Commit:
better spacing of content for affirmations

Commit:
looks amazing on tablet mode

Commit:
dont change button!

fix button on form,

Commit:
add errors to signup form
make validation errors common

Commit:
Updat snapshots

Commit:
Merge branch 'master' into aaronmgdr/collective

Commit:
fix this test which I distnctly remember fixing before

Commit:
personal review

Commit:
revert butto. changes , do in a different PR

Commit:
revert adding Link. (in favor of lightening links PR)

Commit:
remvoe unused style

Commit:
revert order

Commit:
snapper

aaronmgdr commented on packages/web/src/about/About.tsx:
```diff
@@ -145,58 +149,13 @@ export class About extends React.Component<Props & I18nProps> {
   }
 }
 
-const BeautifulQuote = withScreenSize(
```
moved to own file

Commit:
remove files to create separate server PR

jmrossy commented on packages/web/src/alliance/Affirmations.tsx:
```diff
@@ -0,0 +1,152 @@
+function TransItalic({ i18nKey }: { i18nKey: string }) {
+  return (
+    <Trans ns={NameSpaces.alliance} i18nKey={i18nKey}>
+      <Text style={textStyles.italic}>""</Text>
```
nit: not sure why you're using the Trans wrapper here instead of use `t`

aaronmgdr replied:
with `t` its not possible place different styles on parts of a phrase. IE only Italicize one word in a sentence. with `<Trans>` Can do it like `i18nkey: "I would like a <0>lot</0> of bread"  and the the 0th child given to <trans> replaces the `<0>`

jmrossy commented on packages/web/src/alliance/Affirmations.tsx:
```diff
@@ -0,0 +1,152 @@
+        button={{
+          text: t('affirmations.wfpButton'),
+          href:
+            'https://medium.com/celoorg/how-to-design-for-all-stories-from-tanzania-refugees-8b34594d64ae',
```
here you use `celoorg`, below you use `celohq`

aaronmgdr replied:
niice find. gracias.

jmrossy commented on packages/web/src/alliance/Affirmations.tsx:
```diff
@@ -0,0 +1,152 @@
+const SMALLER = 0.8
+
+const mobileStyles = StyleSheet.create({
+  laboratoria: { width: 200 * SMALLER, height: 35 * SMALLER },
```
Nit: should these reference the regular styles? otherwise why have a SMALLER factor, why not just set the numbers directly?

aaronmgdr replied:
you're very right. ha

jmrossy commented on packages/web/src/alliance/Main.tsx:
```diff
@@ -0,0 +1,163 @@
+  },
+]
+
+const blendStyle = { mixBlendMode: 'darken' }
```
why is this out here?

aaronmgdr replied:
oddly whenever blendMode is part of Stylesheet.create it gets lost

jmrossy commented on packages/web/src/alliance/Members.tsx:
```diff
@@ -0,0 +1,238 @@
+  })
+}
+
+const ALL = 'all'
```
should this be part of an enum or something like that?

aaronmgdr replied:
i guess it could be part of the CategoryEnum but then thee will be several places it would need to be filtered out which seems extra complicated.

jmrossy commented on packages/web/src/alliance/Members.tsx:
```diff
@@ -0,0 +1,238 @@
+    // sometimes it is nessessary to break a rule
+    // https://github.com/facebook/react/issues/14326
+    // tslint:disable-next-line: no-floating-promises
+    gatherAllies(setAllies, signal)
```
I had a look at the react link you linked to. This doesn't match the async effect pattern Abramov provided. Why let this promise float?

aaronmgdr replied:
here is my understanding.  `useEffect` is typed to take a function that returns void or a cleanup function. NOT a promise. so cant do await on this as that would require an async function which always return promises.

jmrossy commented on packages/web/src/alliance/PinWheel.tsx:
```diff
@@ -0,0 +1,213 @@
+  },
+})
+
+function getKeyframes() {
```
Nit why is this wrapped in a getter if we don't have params for it? Should this be an array at module scope?

jmrossy commented on packages/web/src/alliance/RisingPinWheelForm.tsx:
```diff
@@ -0,0 +1,44 @@
+
+export default function RisingPinWheelForm() {
+  return (
+    <View style={[standardStyles.darkBackground]}>
```
nit: array of 1

jmrossy commented on packages/web/src/forms/FormComponents.tsx:
```diff
@@ -189,13 +205,93 @@ export function LabeledInput({ name, multiline, hasError, value, onInput, label
+  onPress: (x: any) => void
+}
+
+export function Checkbox({ checked, onPress, name }: CheckboxProps) {
```
Long term, it'd be better for these to each be in their own file to enable better code splitting and lazy loading.

jmrossy commented on packages/web/src/header/Header.3.tsx:
```diff
@@ -164,13 +150,40 @@ export class Header extends React.PureComponent<Props, State> {
   }
 
+  allWhiteLogo = () => {
+    return this.props.router.pathname === menu.ABOUT_US.link && !this.state.belowFoldUpScroll
```
This seems hacky, is there a better way that doesn't hardcode in a pathname?

aaronmgdr replied:
I could define a Set of all path names which should use an all white logo? at module scope

jmrossy commented on packages/web/src/shared/Outbound.tsx:
```diff
@@ -0,0 +1,20 @@
+  )
+}
+
+export function externalizeURL(url: string) {
```
what's this for?

aaronmgdr replied:
if i throw `www.celo.org` straight into a `href` on an anchor its going to direct the person to `celo.org/www.celo.org`.

whereas `//www.celo.org` is known to be a domain. 

this way any URLs set in the airtable will go to the right spot.

jmrossy reviewed (commented):
Overall seems good, just some minor comment below

Commit:
fix article uri
constantize keyframes
array not needed and isa waste of resources.

jmrossy reviewed (approved):
Think you forgot this comment: https://github.com/celo-org/celo-monorepo/pull/3047#discussion_r389437755

But otherwise lgtm

Commit:
Merge branch 'master' into aaronmgdr/collective

Commit:
Add real quotes
Beautiful quote now handles fancy quoting itself,
smaller citation sizes on alliance page,

Commit:
set sizes directly

Commit:
remove Insure

Commit:
add preserve and update desriptions

Commit:
use cGLD is preferred to Celo Gold

Commit:
snapshot on track

Commit:
create mobile specifc layout for affirmations.
add bottom margin on all sizes

Commit:
fix hamburger being camoflaged within the blue banner.

Commit:
nicer rainbow

Commit:
make apply button larger

add test for unfilled submission

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/collective

Commit:
Will now auto public alliance page at 12 noon UTC

Commit:
Revert "Will now auto public alliance page at 12 noon UTC"

This reverts commit bb819c90eda1efaba7b1e8b375c51ddbe73ebf7d.

Commit:
fixes issue with menu rendering on two rows rather than one due to artifacts leftover by dynamic.

key noticing was that it only happened on home page and only on initial render not when changing screens size.

Commit:
make it common to fix it showing key

Commit:
snap

Commit:
Update quotes and ally categories

Commit:
add an form submitted state
increase height as menu otherwise is hidden

Commit:
lighten image to increase legibility of overlayed text

Commit:
oh doh
